### PR TITLE
[ci] change list of docker tags for external use

### PR DESCRIPTION
**Pull request by khluu**
For external use (published on Docker Hub), we remove SHA tag and replace that with `nightly.date.sha` which is generated only on nightly basis

### CODE REVIEW

File: ci/ray_ci/docker_container.py [hunk 1]
```diff
@@ -64,6 +76,13 @@ def _get_image_version_tags(self) -> List[str]:
         return [sha_tag]
 
     def _get_canonical_tag(self) -> str:
+        """
+        Get canonical tag, which is the first tag in the list of tags.
+
+        Args:
+            external: If True, return canonical tag from external image tags.
+                If False, return from the internal image tags.
```
Comment (aslonnie) [inline ci/ray_ci/docker_container.py:84]: out of date doc string?

Comment (khluu) [reply ci/ray_ci/docker_container.py]: yes ... I removed it just now

File: ci/ray_ci/docker_container.py [hunk 2]
```diff
@@ -47,11 +48,22 @@ def __init__(
...
+                internal image tags.
+        """
         branch = os.environ.get("BUILDKITE_BRANCH")
         sha_tag = os.environ["BUILDKITE_COMMIT"][:6]
         pr = os.environ.get("BUILDKITE_PULL_REQUEST", "false")
+        formatted_date = datetime.now().strftime("%y%m%d")
+
         if branch == "master":
+            if external:
+                return [f"nightly.{formatted_date}.{sha_tag}", "nightly"]
```
Comment (aslonnie) [inline ci/ray_ci/docker_container.py:66]: can we remove the `"nightly"` on the non-external case?

and also can we condition returning `nightly` only on `RAYCI_SCHEDULE=nightly` ?

Comment (khluu) [reply ci/ray_ci/docker_container.py:66]: done

**Review by aslonnie — APPROVED** [review]

**Pull request merged**